pkg/cmd/pulumi/package.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+func newPackageCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "package",
+		Short: "Work with Pulumi packages",
+		Long: `Work with Pulumi packages
+
+Subcommands of this command are useful to package authors during development.`,
+		Args: cmdutil.NoArgs,
+	}
+
+	cmd.AddCommand(
+		newExtractSchemaCommand(),
+		newGenSdkCommand(),
+	)
+	return cmd
+}
+
+// schemaFromPackageSource takes a schema source and returns its associated schema. A
+// schema source is either a file (ending with .[json|y[a]ml]) or a plugin with an
+// optional version:
+//
+//		FILE.[json|y[a]ml] | PLUGIN[@VERSION] | PATH_TO_PLUGIN
+//
+func schemaFromSchemaSource(packageSource string) (*schema.Package, error) {
+	var spec schema.PackageSpec
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	sink := cmdutil.Diag()
+	pCtx, err := plugin.NewContext(sink, sink, nil, nil, wd, nil, false, nil)
+	if err != nil {
+		return nil, err
+	}
+	bind := func(spec schema.PackageSpec) (*schema.Package, error) {
+		var loader schema.Loader
+
+		pkg, diags, err := schema.BindSpec(spec, loader)
+		if err != nil {
+			return nil, err
+		}
+		if diags.HasErrors() {
+			return nil, diags
+		}
+		return pkg, nil
+	}
+	if ext := filepath.Ext(packageSource); ext == ".yaml" || ext == ".yml" {
+		f, err := ioutil.ReadFile(packageSource)
```

Comment:
At this point we should be using `os.ReadFile` and `os.WriteFile` rather than the deprecated functions from `ioutil`.